circle.yml

```diff
@@ -0,0 +1,26 @@
+machine:
+  services:
+    - docker
+
+dependencies:
+  pre:
+      # set build status to pending on start
+    - ./util/docker/maketest.sh --pending ${GH_TOKEN}
+      # Login to docker
+    - docker login -e $DOCKER_EMAIL -u $DOCKER_USER -p $DOCKER_PASS
+    - ./util/docker/baseimage/buildbaseimage.sh
+      # Actually the test step, but we want to cache it so it can go here.
+    - ./util/docker/circle/buildcircle.sh
```

Comment:
Caching is only done after the deploy step, so this needs to go here, unfortunately.

The main circeci step will only be green if this passes, if this fails, the execution will stop (sh -c test will never be run, because of the failure)